curvefs/src/client/s3/disk_cache_manager.cpp

```diff
@@ -0,0 +1,365 @@
+        if (ret < 0) {
+            LOG(ERROR) << "create cache dir error. errno = " << errno
+                       << ", dir = " << cacheDir_;
+            return -1;
+        }
+        LOG(INFO) << "cache dir is not exist, create it success."
+                     << ", dir = " << cacheDir_;
+    } else {
+        LOG(INFO) << "cache dir is exist."
+                     << ", dir = " << cacheDir_;
+    }
+
+    ret = CacheWrite_->CreateIoDir(true);
+    if (ret < 0) {
+        LOG(ERROR) << "create cache write dir error. ret = " << ret;
+        return ret;
+    }
+    ret = CacheRead_->CreateIoDir(false);
+    if (ret < 0) {
+        LOG(ERROR) << "create cache read dir error. ret = " << ret;
+        return ret;
+    }
+    LOG(INFO) << "create cache dir sucess.";
+    return 0;
+}
+
+std::string DiskCacheManagerImpl::GetCacheDir() {
+    return cacheDir_;
+}
+
+std::string DiskCacheManagerImpl::GetCacheReadFullDir() {
+    return CacheRead_->GetCacheIoFullDir();
+}
+
+std::string DiskCacheManagerImpl::GetCacheWriteFullDir() {
+    return CacheWrite_->GetCacheIoFullDir();
+}
+
+int64_t DiskCacheManagerImpl::CacheDiskUsedRatio() {
+    struct statfs stat;
+    if (posixWrapper_->statfs(cacheDir_.c_str(), &stat) == -1) {
+        LOG(ERROR) << "get cache disk space error.";
+        return -1;
+    }
+
+    int64_t frsize = stat.f_frsize;
+    int64_t totalBytes = stat.f_blocks * frsize;
+    int64_t freeBytes = stat.f_bfree * frsize;
+    int64_t availableBytes = stat.f_bavail * frsize;
+    int64_t usedBytes = totalBytes - freeBytes;
+    int64_t usedPercent = 100 * usedBytes / (usedBytes + availableBytes) + 1;
+
+    LOG(INFO) << "cache disk usage = " << usedPercent;
+    return usedPercent;
+}
+
+bool DiskCacheManagerImpl::IsDiskCacheFull() {
+    int64_t ratio = CacheDiskUsedRatio();
+    if (ratio < 0) {
+        LOG(ERROR) << "get disk use ratio error.";
+        return false;
+    }
+    if (ratio >= fullRatio_) {
+        LOG(INFO) << "disk cache is full"
+                     << ", ratio = " << ratio
+                     << ", fullRatio = " << fullRatio_;
+        return true;
+    }
+    return false;
+}
+
+bool DiskCacheManagerImpl::IsDiskCacheSafe() {
+    int64_t ratio = CacheDiskUsedRatio();
+    if (ratio < 0) {
+        LOG(ERROR) << "get disk use ratio error." << ratio;
+        return false;
+    }
+    if (ratio < safeRatio_) {
+        LOG(INFO) << "disk cache is safe"
+                     << ", ratio = " << ratio
+                     << ", safeRatio = " << safeRatio_;
+        return true;
+    }
+      LOG(INFO) << "disk cache is not safe"
+                     << ", ratio = " << ratio
+                     << ", safeRatio = " << safeRatio_;
+    return false;
+}
+
+void DiskCacheManagerImpl::TrimCache() {
+    const std::chrono::seconds sleepSec(trimCheckInterval_);
+    LOG(INFO) << "trim function start.";
+    while (sleeper_.wait_for(sleepSec)) {
+        if (!isRunning_) {
```

Comment:
Following handle proccess may be encapsulated as some functions to make the logic be clearer

Reply:
the logical is clear，so add some comments replace